pandas/indexes/interval.py

```diff
@@ -514,13 +511,17 @@ def get_indexer(self, target, method=None, limit=None, tolerance=None):
                 raise ValueError('indexer corresponds to non-unique elements')
             return np.where(start_plus_one == stop, start, -1)
 
+        if isinstance(target, IntervalIndex):
+            raise NotImplementedError(
+                'have not yet implemented get_indexer '
+                'for IntervalIndex indexers')
+
+        if not self.is_unique:
+            indexer, missing = self._engine.get_indexer_non_unique(
+                target.values)
         else:
-            if isinstance(target, IntervalIndex):
-                raise NotImplementedError(
-                    'have not yet implemented get_indexer '
-                    'for IntervalIndex indexers')
-            else:
-                return self._engine.get_indexer(target.values)
+            indexer = self._engine.get_indexer(target.values)
```

Comment:
I think this should go in a separate `Intervalndex.get_indexer_non_unique` method, right?

My understanding of the contract of `Index.get_indexer` is that it returns an integer array with length equal to `target`.